titan/features/haart.py

```diff
@@ -91,26 +93,50 @@ def update_agent(self, model: "model.TITAN"):
             )
             # Go on HAART
             if not self.active:
-                if self.agent.location.params.hiv.haart_cap:
+                if self.agent.location.params.haart.use_cap:
                     # if HAART is based on cap instead of prob, determine number of
                     # HAART agents based on % of diagnosed agents
                     num_dx_agents = self.agent.hiv.dx_counts[self.agent.race][  # type: ignore[attr-defined]
                         self.agent.sex_type
                     ]
                     num_haart_agents = self.counts[self.agent.race][self.agent.sex_type]
 
-                    if num_haart_agents < (haart_params.prob * num_dx_agents):
+                    # take value from dictionary for cap
+                    if num_haart_agents < (haart_params.cap * num_dx_agents):
                         self.initiate(model)
                 else:
-                    if model.run_random.random() < (
-                        haart_params.prob * model.calibration.haart.coverage
-                    ):
-                        self.initiate(model)
+                    if self.ever and self.agent.location.params.haart.reinit:
+                        if model.run_random.random() < haart_params.reinit.prob:
+                            self.initiate(model)
+                    else:
+                        enroll_prob = 0
+                        # Find enroll probability based on time since diagnosis
+                        for i in haart_params.enroll.values():
+                            if i.start <= (model.time - self.agent.hiv.dx_time) < i.stop:  # type: ignore[attr-defined]
```

Comment:
calculate the `model.time - self.agent.hiv.dx_time` outside the loop since that value is static